src/pages/home/report/ReportActionCompose.js

```diff
@@ -166,1137 +179,1146 @@ const getMaxArrowIndex = (numRows, isAutoSuggestionPickerLarge) => {
+            setSuggestionValues((prevState) => ({...prevState, shouldShowMentionSuggestionMenu: false}));
+        }
+    }, [suggestionValues.shouldShowEmojiSuggestionMenu, suggestionValues.shouldShowMentionSuggestionMenu]);
 
     /**
-     * Check if the composer is visible. Returns true if the composer is not covered up by emoji picker or menu. False otherwise.
+     * Determines if we can show the task option
      * @returns {Boolean}
      */
-    checkComposerVisibility() {
-        const isComposerCoveredUp = EmojiPickerActions.isEmojiPickerVisible() || this.state.isMenuVisible || this.props.modal.isVisible;
-        return !isComposerCoveredUp;
-    }
-
-    focusComposerOnKeyPress(e) {
-        const isComposerVisible = this.checkComposerVisibility();
-        if (!isComposerVisible) {
-            return;
-        }
-
-        // If the key pressed is non-character keys like Enter, Shift, ... do not focus
-        if (e.key.length > 1) {
-            return;
+    const taskOption = useMemo(() => {
+        // We only prevent the task option from showing if it's a DM and the other user is an Expensify default email
+        if (!Permissions.canUseTasks(betas) || ReportUtils.isExpensifyOnlyParticipantInReport(report)) {
+            return [];
         }
 
-        // If a key is pressed in combination with Meta, Control or Alt do not focus
-        if (e.metaKey || e.ctrlKey || e.altKey) {
-            return;
-        }
+        return [
+            {
+                icon: Expensicons.Task,
+                text: translate('newTaskPage.assignTask'),
+                onSelected: () => Task.clearOutTaskInfoAndNavigate(reportID),
+            },
+        ];
+    }, [betas, report, reportID, translate]);
 
-        // If the space key is pressed, do not focus
-        if (e.code === 'Space') {
-            return;
-        }
+    /**
+     * Replace the code of emoji and update selection
+     * @param {Number} selectedEmoji
+     */
+    const insertSelectedEmoji = useCallback(
+        (highlightedEmojiIndexInner) => {
+            const commentBeforeColon = value.slice(0, suggestionValues.colonIndex);
+            const emojiObject = suggestionValues.suggestedEmojis[highlightedEmojiIndexInner];
+            const emojiCode = emojiObject.types && emojiObject.types[preferredSkinTone] ? emojiObject.types[preferredSkinTone] : emojiObject.code;
+            const commentAfterColonWithEmojiNameRemoved = value.slice(selection.end);
+
+            updateComment(`${commentBeforeColon}${emojiCode} ${trimLeadingSpace(commentAfterColonWithEmojiNameRemoved)}`, true);
+
+            // In some Android phones keyboard, the text to search for the emoji is not cleared
+            // will be added after the user starts typing again on the keyboard. This package is
+            // a workaround to reset the keyboard natively.
+            if (RNTextInputReset) {
+                RNTextInputReset.resetKeyboardInput(findNodeHandle(textInputRef));
```

Comment:
found root cause of crash. good easy-to-miss case
```suggestion
                RNTextInputReset.resetKeyboardInput(findNodeHandle(textInputRef.current));
```

Reply:
bump on applying this. And then all good!